src/DrDotnet.Profilers/api/clr_profiler_info.rs

```diff
@@ -1220,10 +1220,8 @@ impl CorProfilerInfo3 for ClrProfilerInfo {
         &self,
         expected_completion_milliseconds: u32,
     ) -> Result<(), HRESULT> {
-        
-        match self.get_attached_status() {
-            AttachedStatus::Attached => {
-                self.set_attached_status(AttachedStatus::Detaching);
+        match self.attached_status.compare_exchange(AttachedStatus::Attached as usize, AttachedStatus::Detaching as usize, Ordering::Acquire, Ordering::Relaxed) {
```

Comment:
Right!

Reply:
I still have no clue what Ordering::Something does but that'd be for another day 😬